src/docs/spec-model.md

```diff
@@ -81,3 +81,66 @@ In addition, predicates from the following sources MAY also be encouraged:
+
+Extensions are defined in the `extension_definition` slot of the `MappingSet` object. Each definition is comprised of three elements:
+
+* the name of the slot, as it will appear when used in a mapping set (`slot_name`);
+* a property intended to specify the meaning of the slot (`property`);
+* the type of values expected by the slot (`type_hint`).
+
+A definition MUST have at least a `slot_name`. The name MUST be a XML “non-colonized name” (“NCName”, see [Namespaces in XML, §2](https://www.w3.org/TR/1999/REC-xml-names-19990114/#NT-NCName)). The name MUST NOT match the name of an existing standard slot.
+
+To avoid any conflicy with a future version of the SSSOM specification (which could introduce new standard slot names), implementations are strongly encouraged to craft extension slot names that start with the `ext_` prefix. No new standard slot with a name starting with `ext_` will ever be introduced in any future version of the standard. (This is an advice for SSSOM producers only; SSSOM consumers MUST NOT reject an extension slot solely on the basis that its name does not start with `ext`.)
+
+A definition SHOULD have a `property`. If it does not, implementations MUST automatically construct a default property by catenating the prefix `http://sssom.invalid/` with the name of the extension.
+
+The slot name and the property MUST be unique to each definition. No two definitions can share the same name and/or the same property.
+
+A definition MAY have a `type_hint`. If it does not, a default type of `http://www.w3.org/2001/XMLSchema#string` is assumed.
+
+Once defined, an extension slot may be used as a supplementary slot in either the `Mapping` class or the `MappingSet` class (or both), as if it was a normal, standard slot. How those slots are represented internally and provided to client code is left at the discretion of the implementations.
+
+### Undefined extensions
+
+Undefined extensions are non-standard slots that are not explicitly defined as described in the previous section. Implementations MAY support undefined extensions.
+
+Upon encountering a non-standard slot that is not a defined extension, an implementation that supports undefined extensions MUST behave as if the slot had been defined with:
+
+* a `property` constructed by catenating the prefix `http://sssom.invalid/` to the name of the slot;
+* a `type_hint` of `http://www.w3.org/2001/XMLSchema#string`.
+
+### Restrictions on the values of extension slots
+
+#### General restrictions
+
+The following restrictions apply to all extension slots, regardless of whether they are defined or undefined.
+
+Each mapping set and each mapping can have at most _one_ value for each extension slot. The expected behaviour upon encountering a repeated extension slot is unspecified.
+
+All extension values MUST be representable as literal strings. Complex data structures (e.g., lists or dictionaries) MUST NOT be used.
```

Comment:
Note: _all_ data structures can be represented as a literal string, so not sure if this sentence will do anything. I can have an extension slot, type it as xsd:string, and shove a 10 GB XML document in there if I so want to..

Another way to put it probably is to say that the expected type should be a simple, primitive type, and not a complex, composite type (like list or dict or some such).

Reply:
> Another way to put it probably is to say that the expected type should be a simple, primitive type, and not a complex, composite type (like list or dict or some such).

Agreed.

> I can have an extension slot, type it as xsd:string, and shove a 10 GB XML document in there if I so want to..

Note that as long as it is encoded as a string (as in `my_extension_slot: '<?xml version="1.0"?><article xmlns="http://docbook.org/ns/docbook/">...</article>'), that shouldn’t be a problem at all.

Reply:
Your new docs look perfect.